src/components/TextareaEmojiPicker.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <div class="textarea-emoji-picker">
+    <textarea
+      ref="textarea"
+      class="textarea"
+      :value="value"
+      @input="$emit('input', $event.target.value)"
+    ></textarea>
+    <picker
+      v-show="showEmojiPicker"
+      title="Pick your emoji..."
+      :data="emojiIndex"
+      emoji="point_up"
+      @select="showEmoji"
+    />
+    <span
+      class="emoji-trigger"
+      :class="{ triggered: showEmojiPicker }"
+      @mousedown.prevent="toggleEmojiPicker"
+    >
+      <svg style="width: 20px; height: 20px" viewBox="0 0 24 24">
+        <path
+          fill="#888888"
+          d="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z"
+        />
+      </svg>
+    </span>
+  </div>
+</template>
+
+<script>
+import data from "emoji-mart-vue-fast/data/all.json";
+import "emoji-mart-vue-fast/css/emoji-mart.css";
+
+import { Picker, EmojiIndex } from "emoji-mart-vue-fast/src";
+let emojiIndex = new EmojiIndex(data);
+
+export default {
+  name: "TextareaEmojiPicker",
+  components: { Picker },
+  props: {
+    value: {
+      type: String,
+      default: "",
+    },
+  },
+  data() {
+    return {
+      showEmojiPicker: true,
+      emojiIndex: emojiIndex,
+      emojisOutput: "",
+    };
+  },
+  watch: {
+    emojisOutput(newValue) {
+      console.log(
+        "🚀 ~ file: TextareaEmojiPicker.vue ~ line 56 ~ emojisOutput ~ newValue",
+        newValue
+      );
+    },
+  },
+  methods: {
+    toggleEmojiPicker(emoji) {
+      this.showEmojiPicker = !this.showEmojiPicker;
+      this.emojisOutput = this.emojisOutput + emoji.native;
+    },
+    addEmoji(emoji) {
+      const textarea = this.$refs.textarea;
```

Comment:
This is definitely the main part to deconstruct.  I'll add some explanation comments to the code after this one merges.